flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSink.java

```diff
@@ -374,7 +375,9 @@ public DynamicTableSource.DataStructureConverter createDataStructureConverter(
     }
 
     private Path toStagingPath() {
-        Path stagingDir = new Path(path, ".staging_" + System.currentTimeMillis());
+        // Add a random UUID to prevent multiple sinks from sharing the same staging dir.
+        // Please see FLINK-29114 for more details
+        Path stagingDir = new Path(path, ".staging_" + getStagingPathPostfix(false));
```

Comment:
Why did you decide on creating the `getStagingPathPostfix` method? This creates unnecessary code in the test because we still have to generate the Path there. Alternatively, you could move the entire content of `toStagingPath()` into a static method `generateStagingPath(Path parent, boolean constant)`. That way you just need to call the static method in the test without any path resolution.